### ISSUE TITLE
TypeErrors in RulerViewport and CursorGuide

### ISSUE THREAD

**Issue by mceike** [igvteam/igv.js#1478]
In our embedded setup with IGV.js v2.10.5 (also happens in v2.11.0 but not v2.10.4), we started experiencing multiple uncaught TypeErrors under certain circumstances. I haven't managed to reproduce it elsewhere, but it consistently triggered only after a certain sequence of events and with "Cursor Guide" enabled: 
- Load page with embedded IGV (no errors)
- Navigate to another page for the same analysis
- Return to page with IGV (IGV is redrawn with the same data) --> errors on any mouse move within IGV

Devtools: 

![image](https://user-images.githubusercontent.com/2444948/158833814-ac14964b-3d06-4e25-bae2-b9604ad6adc5.png)

The following changes in `rulerViewport.js`, [patched to v2.11.0](https://github.com/igvteam/igv.js/compare/master...ousamg:fix-rulerViewport-tooltip-undefined), fixed it for us: 

```
diff --git a/js/rulerViewport.js b/js/rulerViewport.js
index ccbd7087..9e0aa453 100644
--- a/js/rulerViewport.js
+++ b/js/rulerViewport.js
@@ -167,7 +167,9 @@ class RulerViewport extends TrackViewport {
                 currentViewport = this
                 this.$tooltip.show()
             } else if (currentViewport.guid !== this.guid) {
-                currentViewport.$tooltip.hide()
+                if (currentViewport.$tooltip) {
+                    currentViewport.$tooltip.hide()
+                }
                 this.$tooltip.show()
                 currentViewport = this
             } else {
@@ -194,7 +196,9 @@ class RulerViewport extends TrackViewport {
 
             // hide tooltip when movement stops
             clearTimeout(timer)
-            timer = setTimeout(() => this.$tooltip.hide(), toolTipTimeout)
+            timer = setTimeout(() => {
+                if (this.$tooltip) this.$tooltip.hide()
+            }, toolTipTimeout)
 
         }
```

Could this be related to #1446?

**Comment by jrobinso** [comment]
@mceike I'm a bit confused as the stack trace you reference wrt "type errors" i snot IGV code.   I think the real problem is the undefined reference.   Will look into this.

**Comment by turner** [comment]
Fixed with commit 83b1b0c6
